src/Microsoft.Identity.Web/WebAppExtensions/MicrosoftIdentityWebAppAuthenticationBuilderExtensions.cs

```diff
@@ -483,5 +489,32 @@ public static partial class MicrosoftIdentityWebAppAuthenticationBuilderExtensio
             options.ForwardSignOut = mergedOptions.ForwardSignOut;
             options.ForwardDefaultSelector = mergedOptions.ForwardDefaultSelector;
         }
+
+        private static void SetIdentityModelLogger(IServiceProvider serviceProvider)
+        {
+            if (serviceProvider != null)
+            {
+                // initialize logger only once
+                if (LogHelper.Logger != NullIdentityModelLogger.Instance)
+                    return;
+                
+                // check if an ILogger was already created by user
+                ILogger logger = serviceProvider.GetService<ILogger<IdentityModelLoggerAdapter>>();
```

Comment:
CategoryName may need to be modified from IdentityModelLoggerAdapter to something more appropriate,